sources/core/Xenko.Core/Threading/ThreadPool.cs

```diff
@@ -17,93 +17,126 @@ namespace Xenko.Core.Threading
     /// </remarks>
     internal class ThreadPool
     {
-        private const long MaxIdleTimeInTicks = 5 * TimeSpan.TicksPerSecond;
+        private const int MaxIdleTimeInMS = 5000;
+        private readonly long MaxIdleTimeTS = (long)((double)Stopwatch.Frequency / 1000 * MaxIdleTimeInMS);
 
         public static readonly ThreadPool Instance = new ThreadPool();
 
+        private readonly Action<object> cachedTaskLoop;
+
         private readonly int maxThreadCount = Environment.ProcessorCount + 2;
         private readonly Queue<Action> workItems = new Queue<Action>();
         private readonly ManualResetEvent workAvailable = new ManualResetEvent(false);
 
         private SpinLock spinLock = new SpinLock();
-        private int workingCount;
-        /// <summary> Usage only within <see cref="spinLock"/> </summary>
+        private int busyCount;
         private int aliveCount;
 
+        public ThreadPool()
+        {
+            // Cache delegate to avoid pointless allocation
+            cachedTaskLoop = (o) => ProcessWorkItems();
```

Comment:
I am not convinced that you are reducing allocation that way. Delegate are usually cached by the runtime, so the previous code using the method group was most likely not creating a delegate at every call.

Do you have profiling supporting your change?

Reply:
Yes, 64 bytes allocated on every call for both ``(o) => ProcessWorkItems()`` and just passing ``ProcessWorkItems`` when ProcessWorkItems' signature has an object parameter.